src/Graby.php

```diff
@@ -745,7 +745,19 @@ private function getSinglePage($html, $url)
         // check it's not what we have already!
         if (false !== $singlePageUrl && $singlePageUrl !== $url) {
             // it's not, so let's try to fetch it...
-            $response = $this->httpClient->fetch($singlePageUrl, false, $siteConfig->http_header);
+            $headers = $siteConfig->http_header;
+
+            $sourceUrl = parse_url($url);
+            $targetUrl = parse_url($singlePageUrl);
+            if (\is_array($sourceUrl)
+                && \is_array($targetUrl)
+                && \array_key_exists('host', $sourceUrl)
+                && \array_key_exists('host', $targetUrl)
+                && $sourceUrl['host'] !== $targetUrl['host']) {
+                $targetSiteConfig = $this->configBuilder->buildForHost($targetUrl['host']);
+                $headers = $targetSiteConfig->http_header;
+            }
```

Comment:
When using multiline in the `if` condition, I usually move the opening brackets on a new line so it's easier to see when the condition list end and the code start.

```suggestion
            if (\is_array($sourceUrl)
                && \is_array($targetUrl)
                && \array_key_exists('host', $sourceUrl)
                && \array_key_exists('host', $targetUrl)
                && $sourceUrl['host'] !== $targetUrl['host'])
            {
                $targetSiteConfig = $this->configBuilder->buildForHost($targetUrl['host']);
                $headers = $targetSiteConfig->http_header;
            }
```

Reply:
Noted, I'll push a fixup

Reply:
php-cs-fixer is reporting an issue with this change